### PR TITLE
Begin adding pagination to catalogue

### PR DESCRIPTION
**Pull request by aswanson-nr**
# Summary
Adds pagination to the catalogue page using a "view more" button.

### CODE REVIEW

**Comment by aswanson-nr** [comment]
The "go to top" button doesn't appear until you are pretty far down the page, could we show this earlier? Maybe when the categories are no longer visible?

**Comment by tabathadelane** [comment]
I was wondering that myself. I will play with some scroll distances.

File: src/pages/index.jsx [hunk 1]
```diff
@@ -355,9 +356,15 @@ const QuickstartsPage = ({ data, location }) => {
...
           >
             {Boolean(search) && <SuperTiles />}
-            {filteredQuickstarts.map((pack) => (
-              <QuickstartTile key={pack.id} featured={false} {...pack} />
-            ))}
+
+            {/* Add pagination grid if no search term or category selected */}
+            {isParamPresent(search) || isParamPresent(category) ? (
+              filteredQuickstarts.map((pack) => (
+                <QuickstartTile key={pack.id} featured={false} {...pack} />
```
Comment (josephgregoryii) [inline src/pages/index.jsx:363]:
```suggestion
                <QuickstartTile key={pack.id} {...pack} />
```

Did we still need `featured={false}`?

**Review by josephgregoryii — DISMISSED** [review]

File: src/pages/index.jsx (hunk 1 above)
Comment (tabathadelane) [reply src/pages/index.jsx:363]: You're right. I think that this is left behind. I couldn't find it referenced in the QuickstartTile component

**Comment by tabathadelane** [comment]
Screen Shot 2022-07-27 at 3 21 28 PM
I'm not sure I love the black on dark gray here with the carousel arrows/guided install background.

File: src/pages/index.jsx [hunk 2]
```diff
@@ -355,9 +356,15 @@ const QuickstartsPage = ({ data, location }) => {
...
             {Boolean(search) && <SuperTiles />}
-            {filteredQuickstarts.map((pack) => (
-              <QuickstartTile key={pack.id} featured={false} {...pack} />
-            ))}
+
+            {/* Add pagination grid if no search term or category selected */}
+            {isParamPresent(search) || isParamPresent(category) ? (
+              filteredQuickstarts.map((pack) => (
+                <QuickstartTile key={pack.id} {...pack} />
+              ))
```
Comment (aswanson-nr) [inline src/pages/index.jsx:364]: I feel a little icky about using this `QuickstartTile` component in two different places to achieve the same result.
This is not required to get this merged, but we could add a `stepSize` field to the `QuickstartGrid` component and then just update that to be the length of the quickstarts array when searching/filtering.
```jsx
<QuickstartGrid quickstarts={filteredQuickstarts} stepSize={isParamPresent(search) || isParamPresent(category) ? filteredQuickstarts.length : 11} />
```

Comment (tabathadelane) [reply src/pages/index.jsx:364]: I like it! much simpler. I'l dive into it tomorrow.

**Review by josephgregoryii — APPROVED** [review]
Looks and feels great! Approving to avoid any blocking after designer approval

**Comment by legnaleama** [comment]
This is looking good! I like the inclusion of the white border around the "Back to the top" CTA, I'll request the team to add it to the Figma library.

Just a couple of adjustments before we push to master:

- [ ] Change color of the string "First step" from `var(--system-text-muted-light` to `#00AC69` which is the "**Green 2**" color on the [Erno palette](https://www.figma.com/file/lopW22jBatqkR5txUtVu1y/%F0%9F%92%AB-Erno-Pattern-Library?node-id=916%3A30114).
- [ ] Change background color of the "See more button" `to transparent` to match the **Secondary button** [style in Erno](https://www.figma.com/file/lopW22jBatqkR5txUtVu1y/%F0%9F%92%AB-Erno-Pattern-Library?node-id=917%3A30676).
- [ ] Remove `hover` effect on the wrapper of the "See more button"

**Comment by tabathadelane** [comment]
> Just a couple of adjustments before we push to master:
> 
> * [x]  Change color of the string "First step" from `var(--system-text-muted-light` to `#00AC69` which is the "**Green 2**" color on the [Erno palette](https://www.figma.com/file/lopW22jBatqkR5txUtVu1y/%F0%9F%92%AB-Erno-Pattern-Library?node-id=916%3A30114).
> * [x]  Change background color of the "See more button" `to transparent` to match the **Secondary button** [style in Erno](https://www.figma.com/file/lopW22jBatqkR5txUtVu1y/%F0%9F%92%AB-Erno-Pattern-Library?node-id=917%3A30676).
> * [x]  Remove `hover` effect on the wrapper of the "See more button"
> 
> @tabathadelane

Thanks for these notes! I have pushed up the changes. @legnaleama 
Screen Shot 2022-08-02 at 10 45 35 AM

Screen Shot 2022-08-02 at 10 46 14 AM

File: src/pages/index.jsx [hunk 3]
```diff
@@ -87,26 +90,53 @@ const QuickstartsPage = ({ data, location }) => {
   const renderGoToTopButton = () => {
```
Comment (aswanson-nr) [inline src/pages/index.jsx:90]: I think this would be a great opportunity to break this out into it's own component.

Comment (tabathadelane) [reply src/pages/index.jsx:90]: Yeah. Good point.

**Review by mickeyryan42 — APPROVED** [review]

**Comment by aswanson-nr** [comment]
Approved

**Review by josephgregoryii — APPROVED** [review]

**Comment by josephgregoryii** [comment]
The back to top button looks soooo much cleaner 👍🏻 great job on this y'all

**Comment by aswanson-nr** [comment]
Got final design sign off ✅

**Pull request merged**